service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -65,52 +67,54 @@ protected ServiceClient(IotHubConnectionString iotHubConnectionString, IotHubSer
             throw new IllegalArgumentException();
         }
 
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_005: [The constructor shall create a SAS token object using the IotHubConnectionString]
-        IotHubServiceSasToken iotHubServiceSasToken = new IotHubServiceSasToken(iotHubConnectionString);
-
         // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_006: [The constructor shall store connection string, hostname, username and sasToken]
         this.iotHubConnectionString = iotHubConnectionString;
         this.hostName = iotHubConnectionString.getHostName();
         this.userName = iotHubConnectionString.getUserString();
-        this.sasToken = iotHubServiceSasToken.toString();
         this.iotHubServiceClientProtocol = iotHubServiceClientProtocol;
+        this.sasTokenExpiryTime = DEFAULT_SAS_TOKEN_EXPIRY_TIME;
+    }
 
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_007: [The constructor shall create a new instance of AmqpSend object]
-        this.amqpMessageSender = new AmqpSend(hostName, userName, sasToken, this.iotHubServiceClientProtocol);
+    /**
+     * Set how long newly generated sas tokens will live for, in seconds. By default, tokens live for 1 year.
+     * A sas token is generated each time a connection is opened. This only applies for connections for sending cloud
+     * to device messages. The file upload notification receiver and feedback message receiver have their own sas token
+     * expiry time that can be set. See {@link com.microsoft.azure.sdk.iot.service.FeedbackMessageListener#setSasTokenExpiryTime(long)} and
+     * {@link com.microsoft.azure.sdk.iot.service.FeedbackMessageListener#setSasTokenExpiryTime(long)}
+     * @param sasTokenExpiryTime the number of seconds newly generated sas tokens will live for.
+     */
+    public void setSasTokenExpiryTime(long sasTokenExpiryTime)
```

Comment:
Does this override the token expiry time calculated by the SDK, if relying on SAS key to generate the auth token?

Reply:
Are file upload notification receiver and feedback message receiver a part of ServiceClient or are they initialized separately? If they are a part of ServiceClient but have their own token expiry time, this API might get confusing.

Reply:
Let's split the feedback receiver and file upload notification receiver out into separate clients to avoid this confusion